### PR TITLE
final binary tree

### PR DESCRIPTION
**Pull request by Manaye**

### CODE REVIEW

**Comment by Manaye** [comment]
### check list

- [x] write method inorder
- [x] write method preorder
- [x] write method postorder
- [x] write method add
- [x] write method search
- [x] test inorder
- [x] test preorder
- [x] test postorder

File: README/Tree.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Trees
```
Comment (mnfmnfm) [inline README/Tree.md:1]: You need a base level README that points at this README.

File: README/Tree.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+2)breadth first: it starts at the root and level by level visits all nodes
+
+## Challenge
+Declaring class node and binary search tree and writing a method
+for traversing a tree with depth first approach like preorder, inorder, postorder
+Write a method named add that adds a new node in the correct location in the binary search tree.
+Write a method named search that brings in a value of node, and returns the node with the desired value.
+
+## Approach & Efficiency
+The run time of depth first or BIG O O(n) time and space
```
Comment (mnfmnfm) [inline README/Tree.md:16]: In what way does this take O(n) space?

File: src/main/java/Tree/BinaryTree.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package Tree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BinaryTree {
+
+    public static List<Node> inOrder(Node root) {
```
Comment (mnfmnfm) [inline src/main/java/Tree/BinaryTree.java:8]: This should return an array, not a list.

File: src/test/java/Tree/BinaryTreeTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            new Node(2),
+            new Node(3),
+            new Node(4),
+            new Node(5),
+            new Node(6),
+            new Node(7));
+
+        List<Node> actual = BinaryTree.inOrder(root);
+        Assert.assertTrue(actual.size() == 7);
+        Assert.assertEquals(actual, expected);
```
Comment (mnfmnfm) [inline src/test/java/Tree/BinaryTreeTest.java:34]: You should test on more than one example input; for instance, this should test for more than just a single binary search tree.

File: src/main/java/Tree/BinarySearchTree.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                add(root.getLeft(), value);
+        } else {
+            if (root.getRight() == null)
+                root.setRight(new Node(value));
+            else
+                add(root.getRight(), value);
+        }
+    }
+
+    public static Node search(Node root, int value) {
```
Comment (mnfmnfm) [inline src/main/java/Tree/BinarySearchTree.java:53]: You don't have any tests for your search method.

File: src/test/java/Tree/BinaryTreeTest.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class BinaryTreeTest {
+
+    @Test
+    public void testInOrder() {
```
Comment (mnfmnfm) [inline src/test/java/Tree/BinaryTreeTest.java:13]: You should test `inOrder` on a basic `Binary Tree`, not on a `BinarySearchTree`.

File: src/main/java/Tree/BinarySearchTree.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
+package Tree;
+
+import java.util.List;
+
+public class BinarySearchTree {
+
+    public static void main(String[] args) {
```
Comment (mnfmnfm) [inline src/main/java/Tree/BinarySearchTree.java:7]: You should not have a main method in any of these files. This is a library, not an application.

File: README/Tree.md (hunk 1 above)
Comment (mnfmnfm) [reply README/Tree.md:1]: Also, why did you create a brand new repo?

File: src/main/java/Tree/BinarySearchTree.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
+package Tree;
+
+import java.util.List;
```
Comment (mnfmnfm) [inline src/main/java/Tree/BinarySearchTree.java:3]: You don't need this import.

**Pull request merged**